lib/spack/spack/test/spec_yaml.py

```diff
@@ -507,3 +508,33 @@ def test_legacy_yaml(tmpdir, install_mockery, mock_packages):
         ("version", "1.2.11"),
     ]
 )
+
+
+@pytest.mark.parametrize(
+    "specfile,expected_hash,reader_cls",
+    [
+        # First version supporting JSON format for specs
+        ("specfiles/hdf5.v013.json", "vglgw4reavn65vx5d4dlqn6rjywnq76d", spack.spec.SpecfileV1),
+        # Introduces full hash in the format, still has 3 hashes
+        ("specfiles/hdf5.v016.json", "stp45yvzte43xdauknaj3auxlxb4xvzs", spack.spec.SpecfileV1),
+        # Introduces "build_specs", see https://github.com/spack/spack/pull/22845
+        ("specfiles/hdf5.v017.json", "xqh5iyjjtrp2jw632cchacn3l7vqzf3m", spack.spec.SpecfileV2),
+        # Use "full hash" everywhere, see https://github.com/spack/spack/pull/28504
+        ("specfiles/hdf5.v019.json", "iulacrbz7o5v5sbj7njbkyank3juh6d3", spack.spec.SpecfileV3),
```

Comment:
The huge numbers of lines changed is mostly due to the data added for this test. The test reads different spec serializations, and check that if they are round-tripped to JSON using the most recent format the hash doesn't change.

Reply:
I've pushed a commit so the data it minified and gzipped

Reply:
:+1: as we agreed in DM